core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp

```diff
@@ -198,6 +198,31 @@ int cuda_get_opt_block_size(const CudaInternal* cuda_instance,
                                 LaunchBounds{});
 }
 
+// Assuming cudaFuncSetCacheConfig(MyKernel, cudaFuncCachePreferL1)
+inline size_t get_shmem_per_sm_prefer_l1(cudaDeviceProp const& properties) {
```

Comment:
There is no way to compute this?

Reply:
Not that I know of

Reply:
Actually there is a `cudaOccSMemPerMultiprocessor()` function in "cuda_occupancy.h" but technically it belongs to implementation details and we are not really supposed to use it.

Reply:
Turns out the function does not do what we want for Volta+
It does not return the smallest possible configuration which is zero...